cmd/apiserver/server_test.go

```diff
@@ -0,0 +1,462 @@
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	calculationsv1 "github.com/vega-project/ccb-operator/pkg/apis/calculations/v1"
+	v1 "github.com/vega-project/ccb-operator/pkg/apis/calculations/v1"
+	"github.com/vega-project/ccb-operator/pkg/client/clientset/versioned/fake"
+	"github.com/vega-project/ccb-operator/pkg/util"
+)
+
+func TestCreateCalculation(t *testing.T) {
+	testCases := []struct {
+		id                 string
+		jsonStr            []byte
+		initialCalculation calculationsv1.Calculation
+		expected           calculationsv1.Calculation
+	}{
+		{
+			id: "one calculation gets created",
+			initialCalculation: calculationsv1.Calculation{
+				TypeMeta:   metav1.TypeMeta{Kind: "Calculation", APIVersion: "vega.io/v1"},
+				ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"created_by_human": "true"}, Name: fmt.Sprintf("calc-%s", util.InputHash([]byte(fmt.Sprintf("%f", 12100.0)), []byte(fmt.Sprintf("%f", 4.0))))},
+				Phase:      calculationsv1.CreatedPhase,
+				Status:     calculationsv1.CalculationStatus{StartTime: metav1.Time{Time: time.Now()}},
+				Spec: calculationsv1.CalculationSpec{Teff: 12100.0, LogG: 4.0, Steps: []v1.Step{
+					{
+						Command: "atlas12_ada",
+						Args:    []string{"s"},
+					},
+					{
+						Command: "atlas12_ada",
+						Args:    []string{"r"},
+					},
+					{
+						Command: "synspec49",
+						Args:    []string{"<", "input_tlusty_fortfive"},
+					},
+				}},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		fakecs := fake.NewSimpleClientset()
+		fakeClient := fakecs.VegaV1()
+
+		var calc struct {
+			Teff string
+			LogG string
+		}
+
+		s := server{
+			logger: logrus.WithField("test-name", "create calculations test"),
+			ctx:    context.Background(),
+			client: fakeClient,
+		}
+
+		tc.expected = tc.initialCalculation
+
+		calc.Teff = fmt.Sprintf("%v", tc.initialCalculation.Spec.Teff)
+		calc.LogG = fmt.Sprintf("%v", tc.initialCalculation.Spec.LogG)
+		data, err := json.Marshal(calc)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		req, err := http.NewRequest("POST", "/calculations/create", bytes.NewBuffer(data))
+		if err != nil {
+			t.Fatal(err)
+		}
+		req.Header.Set("Content-Type", "application/json")
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(s.createCalculation)
+		handler.ServeHTTP(rr, req)
+
+		var actualData *calculationsv1.Calculation
+
+		err = json.Unmarshal(rr.Body.Bytes(), &actualData)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if rr.Code == http.StatusOK {
+			logrus.WithFields(logrus.Fields{"calculation": actualData.Name, "teff": actualData.Spec.Teff, "logG": actualData.Spec.LogG}).Info("Created calculation using the api server...")
+		} else {
+			logrus.Info("No calculation was created...")
+		}
+
+		actualData.Status.StartTime.Time = time.Date(2000, 0, 0, 0, 0, 0, 0, time.UTC)
+		tc.expected.Status.StartTime.Time = time.Date(2000, 0, 0, 0, 0, 0, 0, time.UTC)
```

Comment:
this should be defined in the calculation it. There is no need to change its value in this point.